debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/commands/ToggleInstructionStepModeHandler.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Advantest Europe GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ *******************************************************************************/
+package org.eclipse.cdt.debug.internal.ui.commands;
+
+import org.eclipse.cdt.debug.core.CDebugCorePlugin;
+import org.eclipse.cdt.debug.core.model.ISteppingModeTarget;
+import org.eclipse.cdt.debug.core.model.ITargetProperties;
+import org.eclipse.cdt.debug.internal.ui.actions.CDTDebugPropertyTester;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.Command;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.commands.State;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.IPreferenceChangeListener;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.PreferenceChangeEvent;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.ui.handlers.RegistryToggleState;
+
+/**
+ * Handles the command org.eclipse.cdt.debug.internal.ui.actions.ToggleInstructionStepModeCommand
+ * Turns instruction step mode on/off for selected target.
+ * @author Raghunandana Murthappa
+ */
+public class ToggleInstructionStepModeHandler extends AbstractHandler implements IPreferenceChangeListener {
+
+	private IEclipsePreferences preference = InstanceScope.INSTANCE.getNode("org.eclipse.cdt.dsf.ui"); //$NON-NLS-1$
```

Comment:
This is just a suggestion, feel free do dismiss.
```suggestion
	private IEclipsePreferences preference = InstanceScope.INSTANCE.getNode(CDebugUIPlugin.PLUGIN_ID);
```

Reply:
debug ui plugin has no dependency on DsfUIPlugin. So ignoring this.

Reply:
True, it does not. But the implicit dependency introduced by obtaining preference node with string identifier does not look good for me. What is so DSF-specific in this preference to store it with DSF and not with "plain" CDT Debug?

Reply:
Fixed.